src/Events/Telemetry/RequestFilterProcessor.cs

```diff
@@ -0,0 +1,139 @@
+                    {
+                        activity.SetTag("user.party.id", claim.Value);
+                    }
+                },
+                {
+                    AltinnCoreClaimTypes.AuthenticationLevel,
+                    static (claim, activity) =>
+                    {
+                        activity.SetTag("user.authentication.level", claim.Value);
+                    }
+                },
+                {
+                    AltinnCoreClaimTypes.Org,
+                    static (claim, activity) =>
+                    {
+                        activity.SetTag("user.application.owner.id", claim.Value);
+                    }
+                },
+                {
+                    AltinnCoreClaimTypes.OrgNumber,
+                    static (claim, activity) =>
+                    {
+                        activity.SetTag("user.organization.number", claim.Value);
+                    }
+                },
+                {
+                    "authorization_details",
+                    static (claim, activity) =>
+                    {
+                        SystemUserClaim claimValue = JsonSerializer.Deserialize<SystemUserClaim>(claim.Value);
+                        activity.SetTag("user.system.id", claimValue?.Systemuser_id[0] ?? null);
+                        activity.SetTag("user.system.owner.number", claimValue?.Systemuser_org.ID ?? null);
+                    }
+                },
+            };
+
+            return actions.ToFrozenDictionary();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestFilterProcessor"/> class.
+        /// </summary>
+        public RequestFilterProcessor(IHttpContextAccessor httpContextAccessor = null) : base()
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <summary>
+        /// Determine whether to skip a request
+        /// </summary>
+        public override void OnStart(Activity activity)
+        {
+            bool skip = false;
+            if (activity.OperationName == _requestKind)
+            {
+                skip = ExcludeRequest(_httpContextAccessor.HttpContext.Request.Path.Value);
+            }
+            else if (!(activity.Parent?.ActivityTraceFlags.HasFlag(ActivityTraceFlags.Recorded) ?? true))
+            {
+                skip = true;
+            }
+
+            if (skip)
+            {
+                activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
+            }
+        }
+
+        /// <summary>
+        /// No action on end
+        /// </summary>
+        /// <param name="activity">xx</param>
+        public override void OnEnd(Activity activity)
+        {
+            if (activity.OperationName == _requestKind && _httpContextAccessor.HttpContext is not null)
+            {
+                if (_httpContextAccessor.HttpContext.Request.Headers.TryGetValue("X-Forwarded-For", out StringValues ipAddress))
```

Comment:
We could rename this variable to clientIp.
```suggestion
                if (_httpContextAccessor.HttpContext.Request.Headers.TryGetValue("X-Forwarded-For", out StringValues clientIp))
```